plugins/system/check-ram.rb

```diff
@@ -1,36 +1,63 @@
-#!/usr/bin/env ruby
+#! /usr/bin/env ruby
+#  encoding: UTF-8
 #
-# Check RAM Plugin
+#   check-ram
+#
+# DESCRIPTION:
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#
+# USAGE:
+#
+# NOTES:
+#
+# LICENSE:
+#   Copyright 2012 Sonian, Inc <chefs@sonian.net>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
 #
-
 require 'rubygems' if RUBY_VERSION < '1.9.0'
 require 'sensu-plugin/check/cli'
 
 class CheckRAM < Sensu::Plugin::Check::CLI
-
   option :megabytes,
-    :short  => '-m',
-    :long  => '--megabytes',
-    :description => 'Unless --megabytes is specified the thresholds are in percents',
-    :boolean => true,
-    :default => false
+         short: '-m',
+         long: '--megabytes',
+         description: 'Unless --megabytes is specified the thresholds are in percents',
+         boolean: true,
+         default: false
 
   option :warn,
-    :short => '-w WARN',
-    :proc => proc {|a| a.to_i },
-    :default => 10
+         short: '-w WARN',
+         proc: proc(&:to_i),
+         default: 10
 
   option :crit,
-    :short => '-c CRIT',
-    :proc => proc {|a| a.to_i },
-    :default => 5
+         short: '-c CRIT',
+         proc: proc(&:to_i),
+         default: 5
 
   def run
-    total_ram, free_ram = 0, 0
+    memhash = {}
+    meminfo = File.read('/proc/meminfo')
+    meminfo.each_line do |i|
```

Comment:
Tiny style nit.  You can avoid having to declare the empty `memhash` hash oustide of the block.
```ruby
memhash = meminfo.each_line.with_object({}) do |i, hash|
  key, val = i.split(':')
  val = val.include?('kB') ? val.gsub(/\s+kB/, '') : val
  hash[key.to_s] = val.strip
end
```